### ISSUE TITLE
Drag&Drop pictures work only in Preview in v4.7.0

### ISSUE THREAD

**Issue by ksteitz** [studentquiz/moodle-mod_studentquiz#427]
We encounter a bug using Drag&Drop on image (qtype_ddimageortext). The image shows only in preview; when starting the quiz it cannot be loaded: 

Stack trace: 

    line 2719 of /lib/filelib.php: Error thrown
    line 507 of /mod/studentquiz/lib.php: call to send_stored_file()
    line 2213 of /lib/questionlib.php: call to mod_studentquiz_question_pluginfile()
    line 43 of /question/type/ddimageortext/lib.php: call to question_pluginfile()
    line 5218 of /lib/filelib.php: call to qtype_ddimageortext_pluginfile()
    line 44 of /pluginfile.php: call to file_pluginfile()

There are different picture-paths used in preview and during quiz, e.g. /pluginfile.php/1639588/qtype_ddimageortext/bgimage/1637069/1/6088163/picturename.png (preview and picture correctly loaded) vs. /pluginfile.php/1639588/qtype_ddimageortext/bgimage/1637070/2/6088163/picturename.png (just above error when starting quiz). 

Plugin-Version: v4.7.0 (2022052400) 
Moodle 3.11.11
Database: PostgreSQL 13.8 

Any ideas to fix that issue? 
Thanks a lot!

**Comment by timhunt** [comment]
Yes. I need to releave a 5.0.1 when our current testing is complete.

**Comment by jojoob** [comment]
5.0.1? Will it be compatible with Moodle 3.9 to 3.11? If not, a 4.7.1 would be nice. :)

**Comment by jojoob** [comment]
@ksteitz Meanwhile I just cherry-picked to mentioned commit on top of the MOODLE_3X_STABLE branch (v4.7.0) without changing the version.php: [rzhtwberlin/moodle-mod_studentquiz/tree/fork/MOODLE_3X_STABLE](https://github.com/rzhtwberlin/moodle-mod_studentquiz/tree/fork/MOODLE_3X_STABLE). Since this is just a simple code change you are safe to apply the patched plugin to your Moodle installation without triggering an upgrade.

**Comment by ksteitz** [comment]
Thanks Johannes, seme here. I just applyed the patch on our productive system today in the morning; works fine. 
Nevertheless a 4.7.1 version would be of help, to avoid such manual changes for future updates.
